src/protocol/metadata.rs

```diff
@@ -4,7 +4,64 @@ use error::Result;
 use codecs::{AsStrings, ToByte, FromByte};
 
 use super::{HeaderRequest, HeaderResponse};
-use super::{API_KEY_METADATA, API_VERSION};
+use super::{API_KEY_API_VERSIONS, API_KEY_METADATA, API_VERSION};
+
+#[derive(Debug)]
+pub struct ApiVersionsRequest<'a> {
+    pub header: HeaderRequest<'a>,
+}
+
+impl<'a> ApiVersionsRequest<'a> {
+    pub fn new(correlation_id: i32, client_id: &'a str) -> ApiVersionsRequest<'a> {
+        ApiVersionsRequest {
+            header: HeaderRequest::new(API_KEY_API_VERSIONS,
+                                       API_VERSION,
```

Comment:
According to the Kafka protocol docs, the `ApiVersionsRequest` is only supported by 0.10.0.0 and up. So if you're just hard-coding `API_VERSION` here, which is in turn hard-coded to the Kafka 0.8 API version, then won't this request fail every time?